pact/verifier.py

```diff
@@ -28,68 +28,160 @@ def __str__(self):
+                            'broker version of the participant is mandatory')
+
+    def verify_pacts(
+            self,
+            *pacts,
+            enable_pending=False,
+            include_wip_pacts_since=None,
+            **kwargs
+    ):
+        """Verify pacts from the provider.
+
+        Usage:
+
+        >>> from pact import Verifier
+        >>>
+        >>> verifier = Verifier(
+        ...   provider='UserService',
+        ...   provider_base_url='http://localhost:5001'
+        ... )
+        >>>
+        >>> result, _ = verifier.verify_pacts(
+        ...   'consumer-provider.json',  # Optional
+        ...   enable_pending=True,
+        ... )
+        >>>
+        >>> assert result == 0
+
+        :param pacts: Optional list of pacts to verify. Every pact in that list
+            can be an HTTP URI or a local file.
+        :type pacts: str
+        :param enable_pending: Allow pacts which are in pending state to be
+            verified without causing the overall task to fail.
+        :type enable_pending: bool
+        :param include_wip_pacts_since: Allow pacts that don't match given
+            consumer selectors (or tags) to  be verified, without causing the
+            overall task to fail
+        :type include_wip_pacts_since: str or None
+        :raises Exception: When `publish_verification_results` is set,
+            but `publish_version` is not provided.
+        :return: Returns a tuple of two elements. The first indicates the
+            status of the operation, so that 0 means success, and the second
+            contains the logs of the operation.
+        :rtype: tuple
         """
         self.validate_publish(**kwargs)
 
-        missing_files = [path for path in pacts if not path_exists(path)]
-        if missing_files:
-            raise Exception("Missing pact files {}".format(missing_files))
-
-        pacts = expand_directories(pacts)
+        missing_pacts = [path for path in pacts if not path_exists(path)]
+        if missing_pacts:
+            raise Exception("Missing pact files: {}".format(
+                ', '.join(missing_pacts))
+            )
 
+        pacts = expand_directories(list(pacts))
         options = self.extract_params(**kwargs)
-        success, logs = VerifyWrapper().call_verify(*pacts,
-                                                    provider=self.provider,
-                                                    provider_base_url=self.provider_base_url,
-                                                    enable_pending=enable_pending,
-                                                    include_wip_pacts_since=include_wip_pacts_since,
-                                                    **options)
-
-        return success, logs
 
-    def verify_with_broker(self, enable_pending=False, include_wip_pacts_since=None, **kwargs):
+        return_code, logs = VerifyWrapper().call_verify(
+            *pacts,
```

Comment:
In which case, this one (verify_with_broker) doesn't make sense to have *pacts, because we should use the broker instead.
I don't think it makes sense to try and do both? (verify with files as well as a broker)
(that question probably needs input from someone with wider Pact knowledge)

Reply:
The original request was via URL

https://docs.pact.io/pact_broker/webhooks#events

They didn't mention the file system here